flytekit/core/dynamic_workflow_task.py

```diff
@@ -40,3 +40,35 @@ def __init__(
 dynamic = functools.partial(
     task.task, task_config=_Dynamic(), execution_mode=PythonFunctionTask.ExecutionBehavior.DYNAMIC
 )
+dynamic.__doc__ = """
+Please first see the comments for :py:func:`flytekit.task` and :py:func:`flytekit.workflow`. This ``dynamic``
+concept is an amalgamation of both and enables the user to pursue some :std:ref:`pretty incredible <cookbook:advanced_merge_sort>`
+constructs.
+
+In short, a task's function is run at execution time only, and a workflow function is run at compilation time only (local
+execution notwithstanding). A dynamic workflow is modeled on the backend as a task, but at execution time, the function
+body gets compiled like a workflow, as if the decorator changed from ``@task`` to ``@workflow``. This workflow is
```

Comment:
This is not accurate- if you change task to workflow you cannot use inputs, while in dynamic you can

Reply:
you mean you can't use inputs as if they're native python things?  i'll clarify